mobbex/classes/MobbexHelper.php

```diff
@@ -355,22 +355,19 @@ public static function getInstallments($products)
                 }
             }
 
-            // Check "Ahora" custom fields
-            $categories = Product::getProductCategoriesFull($product['id_product']);
-
-            foreach ($ahora as $key => $value) {
-                // If plan is checked and it's not added yet, add to filter
-                $checked = MobbexCustomFields::getCustomField($product['id_product'], 'product', $key);
-                if ($checked === 'yes' && !in_array('-' . $key, $installments)) {
-                    $installments[] = '-' . $key;
-                    unset($ahora[$key]);
-                } else if (!in_array('-' . $key, $installments)){
-                    foreach($categories as $category){
-                        if (MobbexCustomFields::getCustomField($category['id_category'], 'category', $key) === 'yes') {
-                            $installments[] = '-' . $key;
-                            unset($ahora[$key]);
-                            break;
-                        }
+        }
+
+        // Check "Ahora" custom fields
+        $categoriesId = array();
+        $categoriesId = self::getCategoriesId($products);
+        foreach ($ahora as $key => $value) {
+            //for each key, if it was not added before, then search all categories.
+            if (!in_array('-' . $key, $installments)){
+                foreach($categoriesId as $cat_id){
+                    if (MobbexCustomFields::getCustomField($cat_id, 'category', $key) === 'yes') {
```

Comment:
Puede suceder el caso de que varias categorías tengan checkeado el mismo plan. Para evitar que en ese caso se envíen varias veces los mismos planes habría que frenar el foreach de la variable `$categoriesId` en el momento que se agrega por primera vez

Reply:
El break de la linea 370, no realiza el freno del foreach la primera vez que aparece un plan?, aunque sería correcto usar un "while", de esta forma funciona igualmente.

Reply:
Si, tenés razón! Disculpa!! No lo noté a noche.